work_test.go

```diff
@@ -53,28 +60,27 @@ func TestLockJob(t *testing.T) {
 	if want := int32(0); j.ErrorCount != want {
 		t.Errorf("want ErrorCount=%d, got %d", want, j.ErrorCount)
 	}
-	if j.LastError.Status == pgtype.Present {
+	if j.LastError.Valid {
 		t.Errorf("want no LastError, got %v", j.LastError)
 	}
 
 	// check for advisory lock
 	var count int64
 	query := "SELECT count(*) FROM pg_locks WHERE locktype=$1 AND objid=$2::bigint"
-	if err = j.pool.QueryRow(query, "advisory", j.ID).Scan(&count); err != nil {
+	if err = j.pool.QueryRow(context.Background(), query, "advisory", j.ID).Scan(&count); err != nil {
 		t.Fatal(err)
 	}
 	if count != 1 {
 		t.Errorf("want 1 advisory lock, got %d", count)
 	}
 
 	// make sure conn was checked out of pool
-	stat := c.pool.Stat()
-	total, available := stat.CurrentConnections, stat.AvailableConnections
-	if want := total - 1; available != want {
-		t.Errorf("want available=%d, got %d", want, available)
-	}
+	stat = c.pool.Stat()
+	require.Equal(t, int32(5), stat.MaxConns())
+	require.Equal(t, int32(2), stat.TotalConns())
+	require.Equal(t, int32(1), stat.IdleConns())
```

Comment:
@alexgo84 I am not 100% sure about those tests. So before we checked if the available connections were `total - 1` to make sure that we got one connection out of the pool.
But the "available" stat is not there anymore so I replaced it with this check. 

From my investigation it looks like this:
1. In the beginning we have Max-Connections of 5 and Total and Idle is 0
2. When the first DB-Connection is requested it opens 2 connections, 1 to return and 1 is idle (I guess this is done so the next connection-request is faster, so we always have one idle connection to give back)

does that make sense?